jsinterop-ts-defs-processor/src/main/java/com/vertispan/tsdefs/model/TsNullableType.java

```diff
@@ -0,0 +1,44 @@
+/*
+ * Copyright © 2023 Vertispan
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.vertispan.tsdefs.model;
+
+public class TsNullableType extends TsType implements IsNullableTsType {
+
+  private final TsType type;
+  private boolean undefined = true;
+
+  public TsNullableType(TsType type) {
+    super(type.name, type.namespace);
+    this.type = type;
+  }
+
+  public static TsNullableType of(TsType tsType) {
+    return new TsNullableType(tsType);
```

Comment:
maybe check if input is already instanceof TsNullableType, if so just return it without wrapping again?